app/scripts/modules/core/src/presentation/Markdown.tsx

```diff
@@ -1,40 +1,41 @@
 import * as React from 'react';
 import { HtmlRenderer, Parser } from 'commonmark';
 import * as DOMPurify from 'dompurify';
+import './Markdown.less';
 
 export interface IMarkdownProps {
   [key: string]: any;
 
   /** markdown */
   message: string;
+
   /** optional tag */
   tag?: string;
+
+  /** The className(s) to apply to the tag (.Markdown class is always applied) */
+  className?: string;
 }
 
 /**
  * Renders markdown into a div (or some other tag)
  * Extra props are passed through to the rendered tag
  */
-export class Markdown extends React.Component<IMarkdownProps> {
-  public static defaultProps: Partial<IMarkdownProps> = {
-    tag: 'div',
-  };
-
-  private parser: Parser = new Parser();
-  private renderer: HtmlRenderer = new HtmlRenderer();
+export function Markdown(props: IMarkdownProps) {
+  const { message, tag: tagProp, className: classNameProp, ...rest } = props;
+  const tag = tagProp || 'div';
 
-  public render() {
-    const { message, tag, ...rest } = this.props;
+  const parser = React.useMemo(() => new Parser(), []);
+  const renderer = React.useMemo(() => new HtmlRenderer(), []);
+  const className = `Markdown ${classNameProp || ''}`;
 
-    if (message == null) {
-      return null;
-    }
+  if (message == null) {
+    return null;
+  }
 
-    const restProps = rest as React.DOMAttributes<any>;
-    const parsed = this.parser.parse(message.toString());
-    const rendered = this.renderer.render(parsed);
-    restProps.dangerouslySetInnerHTML = { __html: DOMPurify.sanitize(rendered) };
+  const restProps = rest as React.DOMAttributes<any>;
```

Comment:
this makes me wish there was a nice way to describe `IMarkdownProps` as a union of `React.DOMAttributes<any>` and the other fields in a way where using `...rest` would automatically let the type system narrow back down to `React.DOMAttributes<any>`. Oh well.